modules/natgw/main.tf

```diff
@@ -0,0 +1,96 @@
+resource "azurerm_public_ip" "this" {
+  count = (var.create_natgw && var.create_pip) ? 1 : 0
+
+  name                = "${var.name}-pip"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  allocation_method   = "Static"
+  sku                 = "Standard"
+  zones               = var.zone != null ? [var.zone] : null
+
+  tags = var.tags
+}
+
+data "azurerm_public_ip" "this" {
+  count = (var.create_natgw && !var.create_pip && var.existing_pip_name != null) ? 1 : 0
+
+  name                = var.existing_pip_name
+  resource_group_name = var.existing_pip_resource_group_name == null ? var.resource_group_name : var.existing_pip_resource_group_name
+}
+
+resource "azurerm_public_ip_prefix" "this" {
+  count = (var.create_natgw && var.create_pip_prefix) ? 1 : 0
+
+  name                = "${var.name}-pip-prefix"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  ip_version          = "IPv4"
+  prefix_length       = var.pip_prefix_length
+  sku                 = "Standard"
+  zones               = var.zone != null ? [var.zone] : null
+
+  tags = var.tags
+}
+
+data "azurerm_public_ip_prefix" "this" {
+  count = (var.create_natgw && !var.create_pip_prefix && var.existing_pip_prefix_name != null) ? 1 : 0
+
+  name                = var.existing_pip_prefix_name
+  resource_group_name = var.existing_pip_prefix_resource_group_name == null ? var.resource_group_name : var.existing_pip_prefix_resource_group_name
```

Comment:
Could we make it simpler e.g. ``coalesce(var.existing_pip_prefix_resource_group_name, var.resource_group_name)`` ?

Reply:
yes, why not. There were some problems with `coalesce` though, can't remember details.

Reply:
if there are problems with `coalesce`, let's leave it as it's

Reply:
:+1: did coalesce as suggested